test/supplemental_genbank_tests/genbank_upload_full_test.py

```diff
@@ -152,18 +192,340 @@ def test_simple_shock_upload(self):
         ### Test for upload from SHOCK - upload the file to shock first
         print('attempting upload through shock')
         gbk_path = "data/e_coli/GCF_000005845.2_ASM584v2_genomic.gbff"
-        data_file_cli = DataFileUtil(os.environ['SDK_CALLBACK_URL'])
         shutil.copy(gbk_path, self.cfg['scratch'])
-        shock_id = data_file_cli.file_to_shock({
+        shock_id = self.dfuClient.file_to_shock({
             'file_path': os.path.join(self.cfg['scratch'], gbk_path.split("/")[-1])
         })['shock_id']
         print("Running test")
         ws_obj_name2 = 'MyGenome.2'
-        result = self.getImpl().genbank_to_genome(self.getContext(), {
+        result = self.serviceImpl.genbank_to_genome(self.ctx, {
                 'file': {'shock_id': shock_id},
-                'workspace_name': self.getWsName(),
+                'workspace_name': self.wsName,
                 'genome_name': ws_obj_name2,
             })[0]
         self.assertIsNotNone(result['genome_ref'])
         self.assertTrue(int(
             result['genome_info'][10]['Number of Protein Encoding Genes']) > 0)
+
+    def _run_test_fail(self, params, error_message):
+        with pytest.raises(Exception) as got:
+            self.serviceImpl.genbank_to_genome(self.ctx, params)
+        assert_exception_correct(got.value, ValueError(error_message))
+
+    def _run_test_fail_mass(self, params, error_message):
+        with pytest.raises(Exception) as got:
+            self.serviceImpl.genbanks_to_genomes(self.ctx, params)
+        assert_exception_correct(got.value, ValueError(error_message))
+
+    def _load_expected_data(self, json_path):
+        with open(json_path, "r") as read_file:
+            data = json.load(read_file)
+        return data
+
+    def _retrieve_provenance(self, provenance):
+        # make a copy to avoid modifying the original provenance
+        provs = [prov.copy() for prov in provenance]
+        for prov in provs:
+            for key in ["time", "epoch"]:
+                prov.pop(key)
+            for subaction in prov['subactions']:
+                subaction.pop("commit")
+        return provs
+
+    def _retrieve_genome_metadata(self, metadata):
+        # make a copy to avoid modifying the original metadata
+        metadata = dict(metadata)
+        metadata.pop("Assembly Object")
+        return metadata
+
+    def _retrieve_genome_data(self, data):
+        # make a copy to avoid modifying the original genome data
+        data = dict(data)
+        for key in ["assembly_ref", "genbank_handle_ref"]:
+            data.pop(key)
+        for key in ["cdss", "features", "mrnas", "non_coding_features"]:
+            for dist in data.get(key):
+                dist.pop("aliases", None)
```

Comment:
Why do aliases need to be discarded?

Reply:
Aliases order is not constant. I had an ordered function that will sort them and compare, but thought was overkill.
<img width="1273" alt="Screenshot 2024-04-22 at 3 27 47 PM" src="https://github.com/kbaseapps/GenomeFileUtil/assets/22202096/6665acae-bc7d-41a0-a033-4553714b7d44">

Reply:
Just put them in a set and compare

Reply:
Is this ok? It worked
```
def ordered(obj):
    if isinstance(obj, dict):
        return sorted((k, ordered(v)) for k, v in obj.items())
    if isinstance(obj, list):
        return sorted(ordered(x) for x in obj)
    if isinstance(obj, int):
        return str(obj)
    else:
        return obj
```

Reply:
I agree with your originally assessment, that seems like overkill and is changing things all through the data structure. I would only do very targeted tweaks when testing ordering, like using a set for a field if necessary

Reply:
:+1: